pyne/mcnp.py

```diff
@@ -1990,210 +1987,235 @@ def _read_meshtal_head(self, f):
-                    self.tally[tally_num] = MeshTally(f, tally_num,
+                    self.tally[tally_num] = self.create_meshtally(f, tally_num,
                                                       self.tags[tally_num],
                                                       mesh_has_mats=self._meshes_have_mats)
                 else:
-                    self.tally[tally_num] = MeshTally(f, tally_num,
+                    self.tally[tally_num] = self.create_meshtally(f, tally_num,
                                                       mesh_has_mats=self._meshes_have_mats)
 
             line = f.readline()
 
+    def create_meshtally(self, f, tally_number, tag_names=None,
+        mesh_has_mats=False):
+        """
+        This function creates a Mesh instance from MCNP meshtal file.
 
-class MeshTally(StatMesh):
-    """This class stores all information from all single MCNP mesh tally that
-    exists within some meshtal file. Header information is stored as attributes
-    and the "mesh" attribute is a MOAB mesh with all result and relative error
-    data tagged. This class inherits from StatMesh, exposing all statistical
-    mesh manipulation methods.
-
-    Attributes
-    ----------
-    tally_number : int
-        The MCNP tally number. Must end in 4 (e.g. 4, 14, 214).
-    particle : string
-        Either "neutron" for a neutron mesh tally or "photon" for a photon mesh
-        tally.
-    dose_response : bool
-        True is the tally is modified by a dose response function.
-    x_bounds : list of floats
-        The locations of mesh vertices in the x direction.
-    y_bounds : list of floats
-        The locations of mesh vertices in the y direction.
-    z_bounds : list of floats
-        The locations of mesh vertices in the z direction.
-    e_bounds : list of floats
-        The minimum and maximum bounds for energy bins
-    mesh :
-        An PyMOAB core instance tagged with all results and
-        relative errors
-    tag_names : iterable
-        Four strs that specify the tag names for the results, relative errors,
-        total results, and relative errors of the total results.
-
-    Notes
-    -----
-    All Mesh/StatMesh attributes are also present via a super() call to
-    StatMesh.__init__().
-
-    """
-
-    def __init__(self, f, tally_number, tag_names=None, mesh_has_mats=False):
-        """Create MeshTally object from a filestream open to the second
-        line of a mesh tally header (the neutron/photon line). MeshTally objects
-        should be instantiated only through the Meshtal class.
-
-        Parameters
-        ----------
-        f : filestream
-            Open to the neutron/photon line.
+        Parameters:
+        -----------
+        f : str or filestream
+            Filestream of the meshtal file.
         tally_number : int
             The MCNP fmesh4 tally number (e.g. 4, 14, 24).
+        particle : str
+            The particle type, 'neutron' or 'photon'.
```

Comment:
Is this an attribute here or an attribute of the MeshTally class?

Reply:
These two lines are deleted in https://github.com/pyne/pyne/pull/1199/commits/bae3bf1d7e03f507e8729edd3197c8daf20853e5, because it is not an input parameter. The `particle` is an attribute of the `MeshTally` class, but not the `Meshtal`. In theory, a `meshtal` file could contain meshtallies of different particles.